src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,130 @@
+import * as tree from './tree_methods'
+
+let root;
+let note1;
+let note2;
+let note3;
+let note21;
+let note22;
+let note221;
+let note222;
+let note31;
+
+beforeEach(() => {
```

Comment:
Can we make the tree once and than copy it? See: prototype pattern

Reply:
Will do